http_crawler/crawler.go

```diff
@@ -119,3 +121,36 @@ func contains(haystack []int, needle int) bool {
 
 	return false
 }
+
+func IsAllowedHost(needle string, allowedHosts []*url.URL) bool {
```

Comment:
Is there a reason why these two functions are [publicly exported](https://golang.org/ref/spec#Exported_identifiers) and why they don't have tests?